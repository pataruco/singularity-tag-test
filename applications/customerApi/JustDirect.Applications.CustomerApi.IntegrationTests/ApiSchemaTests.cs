using HotChocolate.Execution;
using JustDirect.Applications.CustomerApi.Queries;
using Microsoft.Extensions.DependencyInjection;
using Snapshooter.NUnit;

namespace JustDirect.Applications.CustomerApi.IntegrationTests;

public class ApiSchemaTests
{
    [Test]
    public void SchemaChangeTest()
    {
        // arrange
        var schema = new ServiceCollection()
            .AddGraphQLServer()
            .AddQueryType<Query>()
            .AddMutationType<Mutation>()
            .AddGlobalObjectIdentification()
            .BuildSchemaAsync();
        schema.ToString().MatchSnapshot();
    }
}