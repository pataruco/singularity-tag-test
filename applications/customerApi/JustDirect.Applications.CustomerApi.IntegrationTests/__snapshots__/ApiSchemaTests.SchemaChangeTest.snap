schema {
  query: Query
  mutation: Mutation
}

"The node interface is implemented by entities that have a global unique identifier."
interface Node {
  id: ID!
}

type Customer implements Node {
  id: ID!
  userId: ID
  contactId: ID!
  email: String!
  salutation: String
  firstName: String
  lastName: String
}

type Mutation {
  updateOneCustomer(where: CustomerWhereUniqueInput! input: CustomerInput!): Customer
}

type Query {
  "Fetches an object given its ID."
  node("ID of the object." id: ID!): Node @cost(weight: "10")
  "Lookup nodes by a list of IDs."
  nodes("The list of node IDs." ids: [ID!]!): [Node]! @cost(weight: "10")
  customer(where: CustomerWhereUniqueInput!): Customer
  test(request: HelloRequestInput!): String! @cost(weight: "10")
}

input CustomerInput {
  salutation: String
}

input CustomerWhereUniqueInput {
  id: ID
  userId: ID
  contactId: ID
}

input HelloRequestInput {
  name: String
}

"The purpose of the `cost` directive is to define a `weight` for GraphQL types, fields, and arguments. Static analysis can use these weights when calculating the overall cost of a query or response."
directive @cost("The `weight` argument defines what value to add to the overall cost for every appearance, or possible appearance, of a type, field, argument, etc." weight: String!) on SCALAR | OBJECT | FIELD_DEFINITION | ARGUMENT_DEFINITION | ENUM | INPUT_FIELD_DEFINITION
